src/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.Escaping.cs

```diff
@@ -79,13 +79,16 @@ public static int NeedsEscaping(ReadOnlySpan<byte> value, JavaScriptEncoder enco
             return idx;
         }
 
-        public static int NeedsEscaping(ReadOnlySpan<char> value, JavaScriptEncoder encoder)
+        public static unsafe int NeedsEscaping(ReadOnlySpan<char> value, JavaScriptEncoder encoder)
         {
             int idx;
 
             if (encoder != null)
             {
-                idx = encoder.FindFirstCharacterToEncodeUtf8(MemoryMarshal.Cast<char, byte>(value));
+                fixed (char* ptr = value)
+                {
+                    idx = encoder.FindFirstCharacterToEncode(ptr, value.Length);
```

Comment:
Some implementations of the `FindFirstCharacterToEncode` method may not accept null pointers. We should special-case `value.IsEmpty` at the beginning of this method and bail early.

Reply:
Will fix this and add a buggy javascriptencoder implementation as a test.